### PR TITLE
Show a summary when adding or removing variation option values

### LINKED ISSUE #39688: Show a summary when adding or removing variation option values
It'd be handy to display a simple summary when the user adds or removes values in the edit variation option modal.

**Design**

AkVNGImLgSqCObTQ3idVn7-fi-3301_326789 (temporary link)

image

**Acceptance criteria**

- [ ] When the user adds values, the message should say: `X variations will be added`.
- [ ] When the user removes values, the message should say: `Y variations will be removed`.
- [ ] When the user does both, the message should say: `X variations will be added, Y variations will be removed`.
- [ ] We calculate the numbers dynamically. If necessary, we show a spinner inside the message area (and hide the text).
- [ ] The message is not shown when there are no changes to the option's values.

### LINKED ISSUE #40512: [Product Block Editor] Variations are not generated automatically if any of the previous variations has the price set
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In the New Product Editor, variations should be generated automatically when variations options are created, as indicated in the testing instructions of [this PR](https://github.com/woocommerce/woocommerce/pull/40343).

However, in that PR, testing fails in the 8th step. When you add variation options in a draft product, variations are generated automatically, which is great. The variations are created without any price and you're suggested to add it. However, if you set the price to any of these variations and add more variation options, no more variations are created automatically and some errors appear in the JS console.

These errors are related to a HTTP 400 API response to the following request:
`/wp-json/wc/v3/products/15/variations/generate?_locale=user`.

This is the response
```
{
    "code": "rest_invalid_param",
    "message": "Invalid parameter(s): default_values",
    "data": {
        "status": 400,
        "params": {
            "default_values": "default_values[stock_quantity] is not of type integer."
        },
        "details": {
            "default_values": {
                "code": "rest_invalid_type",
                "message": "default_values[stock_quantity] is not of type integer.",
                "data": {
                    "param": "default_values[stock_quantity]"
                }
            }
        }
    }
}
```
This impedes further variations to be created automatically when new variation options are created.

### Expected behavior

Variations should be generated automatically.

### Actual behavior

Variations are not created automatically.

### Steps to reproduce

1. Enable the new product editor in WooCommerce > Settings > Advanced > Features
2. Go to WooCommerce > Products > Add new
3. Add a unique name and click save draft
4. Now go to the Variations tab and add a couple variation options.
5. Variations should be auto generated
6. Now add a price to either all of them or at-least the first one.
7. Add some new variation options
The new variations should be auto generated, Note: the new variations should include the regular price of the first existing variation by default.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #39688
Closes #40512

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure feature `New product editor` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Make sure feature `product-variation-management` is enabled under `Features` tab from `/wp-admin/tools.php?page=woocommerce-admin-test-helper` (WooCommerce Beta Tester plugin) is required
3. Go to `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations` and add some attributes.
4. Then go to edit one attribute from the list.
5. If you add values, the message should say: `X variations will be added.`
6. If you remove values, the message should say: `Y variations will be removed.`
7. If you do both, the message should say: `X variations will be added, Y variations will be removed.`

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/attribute-control/edit-attribute-modal.tsx [hunk 1]
```diff
@@ -185,6 +250,10 @@ export const EditAttributeModal: React.FC< EditAttributeModalProps > = ( {
 							</div>
 						) }
 					</div>
+
+					{ Boolean( additions || deletions ) && (
+						<Notice>{ getNoticeMessage() }</Notice>
+					) }
```
Comment (octaedro) [inline packages/js/product-editor/src/components/attribute-control/edit-attribute-modal.tsx:261]: I believe that if we don't verify that we're under the `Variations` tab, there's a chance we might display the message while editing `Attributes` in the `Organization` tab

Screen Capture on 2023-09-28 at 15-19-25

**Review by octaedro — COMMENTED** [review]
Great job @mdperez86! The code looks good; I just left a small comment.

File: packages/js/product-editor/src/components/attribute-control/edit-attribute-modal.tsx (hunk 1 above)
Comment (mdperez86) [reply packages/js/product-editor/src/components/attribute-control/edit-attribute-modal.tsx:261]: Ohh!! good catch. Let me fix this asap!

**Review by octaedro — APPROVED** [review]
Good job @mdperez86! LGTM 🚀

**Pull request merged**